packages/coding-standards/ecs.php

```diff
@@ -329,63 +314,43 @@
             'while',
             'yield',
         ],
-    ]);
+    ])
     // Slevomat Coding Standards
-    $ecsConfig->ruleWithConfiguration(\SlevomatCodingStandard\Sniffs\Classes\RequireMultiLineMethodSignatureSniff::class, [
+    ->withConfiguredRule(RequireMultiLineMethodSignatureSniff::class, [
         'minLineLength' => 120,
-    ]);
-    $ecsConfig->ruleWithConfiguration(DeclareStrictTypesSniff::class, [
+    ])
+    ->withConfiguredRule(DeclareStrictTypesSniff::class, [
         'spacesCountAroundEqualsSign' => 0,
-    ]);
-    $ecsConfig->rule(RequireTrailingCommaInDeclarationSniff::class);
-    $ecsConfig->rule(RequireTrailingCommaInClosureUseSniff::class);
-    $ecsConfig->rule(RequireTrailingCommaInCallSniff::class);
-    $ecsConfig->rule(TrailingArrayCommaSniff::class);
-    $ecsConfig->rule(RequireConstructorPropertyPromotionSniff::class);
-    $ecsConfig->rule(PropertyTypeHintSniff::class);
-    $ecsConfig->rule(DisallowEqualOperatorsSniff::class);
-    $ecsConfig->rule(ValidClassNameSniff::class);
-    $ecsConfig->rule(NoUselessElseFixer::class);
-    $ecsConfig->rule(AssignmentInConditionSniff::class);
-    $ecsConfig->rule(DisallowEmptySniff::class);
-    $ecsConfig->ruleWithConfiguration(EarlyExitSniff::class, [
+    ])
+    ->withConfiguredRule(EarlyExitSniff::class, [
         'ignoreStandaloneIfInScope' => true,
         'ignoreOneLineTrailingIf' => true,
         'ignoreTrailingIfWithOneInstruction' => true,
-    ]);
-    $ecsConfig->rule(ParentCallSpacingSniff::class);
-    $ecsConfig->ruleWithConfiguration(ReferenceUsedNamesOnlySniff::class, [
+    ])
+    ->withConfiguredRule(ReferenceUsedNamesOnlySniff::class, [
         'allowPartialUses' => true,
-    ]);
-    $ecsConfig->ruleWithConfiguration(DocCommentSpacingSniff::class, [
+    ])
+    ->withConfiguredRule(DocCommentSpacingSniff::class, [
         'linesCountBetweenDifferentAnnotationsTypes' => 0,
-    ]);
-    $ecsConfig->rule(UselessIfConditionWithReturnSniff::class);
-
+    ])
     // Code Metrics
-    $ecsConfig->ruleWithConfiguration(ClassLengthSniff::class, [
+    ->withConfiguredRule(ClassLengthSniff::class, [
         'maxLinesLength' => 550,
-    ]);
-    $ecsConfig->ruleWithConfiguration(FunctionLengthSniff::class, [
+    ])
+    ->withConfiguredRule(FunctionLengthSniff::class, [
         'maxLinesLength' => 60,
-    ]);
-
-    $ecsConfig->skip([
-        __DIR__ . '/tests/Unit/**/wrong/*',
-        __DIR__ . '/tests/Unit/**/Wrong/*',
-        __DIR__ . '/tests/Unit/**/correct/*',
-        __DIR__ . '/tests/Unit/**/Correct/*',
-        __DIR__ . '/tests/Unit/**/fixed/*',
+    ])
+    ->withSkip([
```

Comment:
could we please discuss this briefly? Is not there a problem that project-base includes the config from `shopsys/coding-standards` and defines its own `withSkip`?

Reply:
skips are not a problem in general, but the problem is when some package defines skip and another has the same rule skipped with other files.
e.g., 

```
// project-base/ecs.php

->withSkip([
    FunctionLength::class => [
        __DIR__ . '/file1',
        __DIR__ . '/file2',
    ],
    OneSniff::class,
]);

// monorepo ecs.php

->withSets([
    __DIR__ . '/project-base/ecs.php',
])
->withSkip([
    FunctionLength::class => [
        __DIR__ . '/file3',
        __DIR__ . '/file4',
    ],
    TwoSniff::class,
]);

```

Both OneSniff and TwoSniff are skipped, but the list of files for FunctionLength is not merged, but overridden.